code/modules/mob/living/carbon/xenomorph/castes/puppeteer/abilities_puppeteer.dm

```diff
@@ -0,0 +1,449 @@
+	RegisterSignal(new_puppet, list(COMSIG_MOB_DEATH, COMSIG_QDELETING), PROC_REF(remove_puppet))
+	RegisterSignal(new_puppet, COMSIG_XENOMORPH_POSTATTACK_LIVING, PROC_REF(postattack)) //harvesting claws
+	puppets += new_puppet
+
+/// Cleans up puppet from our list
+/datum/action/xeno_action/activable/refurbish_husk/proc/remove_puppet(datum/source)
+	SIGNAL_HANDLER
+	puppets -= source
+	UnregisterSignal(source, list(COMSIG_MOB_DEATH, COMSIG_QDELETING, COMSIG_XENOMORPH_POSTATTACK_LIVING))
+
+/datum/action/xeno_action/activable/refurbish_husk/proc/postattack(mob/living/source, useless, damage)
+	var/mob/living/carbon/xenomorph/owner_xeno = owner
+	owner_xeno.plasma_stored = min(owner_xeno.plasma_stored + round(damage / 0.9), owner_xeno.xeno_caste.plasma_max)
+
+// ***************************************
+// *********** Stitch Puppet
+// ***************************************
+/datum/action/xeno_action/activable/puppet
+	name = "Stitch Puppet"
+	action_icon_state = "stitch_puppet"
+	desc = "Uses 350 biomass to create a flesh homunculus to do your bidding, at an adjacent target location."
+	plasma_cost = 350
+	cooldown_timer = 25 SECONDS
+	target_flags = XABB_TURF_TARGET
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_PUPPET,
+	)
+
+/datum/action/xeno_action/activable/puppet/can_use_ability(atom/target, silent = FALSE, override_flags)
+	. = ..()
+	var/mob/living/carbon/xenomorph/owner_xeno = owner
+	if(isclosedturf(target))
+		if(!silent)
+			target.balloon_alert(owner_xeno, "THATS A FUCKING WALL")
+		return FALSE
+
+	var/datum/action/xeno_action/activable/refurbish_husk/huskaction = owner.actions_by_path[/datum/action/xeno_action/activable/refurbish_husk]
+	if(length(huskaction.puppets) >= owner_xeno.xeno_caste.max_puppets)
+		owner_xeno.balloon_alert(owner_xeno, "cant have more than [owner_xeno.xeno_caste.max_puppets]!")
+		return FALSE
+
+	if(!owner_xeno.Adjacent(target))
+		if(!silent)
+			to_chat(owner_xeno, span_xenonotice("We need to be next to where we want to create a puppet."))
+		return FALSE
+
+	if(!.)
+		return
+	owner_xeno.face_atom(target)
+	//reverse gib here
+	owner_xeno.visible_message(span_warning("[owner_xeno] begins to vomit out biomass and skillfully sews various bits and pieces together!"))
+	if(!do_after(owner_xeno, 8 SECONDS, FALSE, target, BUSY_ICON_CLOCK, extra_checks = CALLBACK(owner_xeno, TYPE_PROC_REF(/mob, break_do_after_checks), list("health" = owner_xeno.health))))
+		//cancel effects
+		return FALSE
+	//cancel effects
+	owner_xeno.visible_message(span_warning("[owner_xeno] forms a repulsive puppet!"))
+	succeed_activate()
+
+/datum/action/xeno_action/activable/puppet/use_ability(atom/target)
+	var/turf/target_turf = get_turf(target)
+
+	var/datum/action/xeno_action/activable/refurbish_husk/huskaction = owner.actions_by_path[/datum/action/xeno_action/activable/refurbish_husk]
+	huskaction.add_puppet(new /mob/living/carbon/xenomorph/puppet(target_turf, owner))
```

Comment:
You should copy the tts voice for the latter say() that can play

Reply:
i play with ttss off

Reply:
Thats not what I said

Reply:
did